pkg/volume/csi/csi_mounter.go

```diff
@@ -235,6 +235,19 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	}
 	volAttribs = mergeMap(volAttribs, serviceAccountTokenAttrs)
 
+	driverSupportsCSIVolumeMountGroup := false
+	var nodePublishFSGroupArg *int64
+	if utilfeature.DefaultFeatureGate.Enabled(features.DelegateFSGroupToCSIDriver) {
+		driverSupportsCSIVolumeMountGroup, err = csi.NodeSupportsVolumeMountGroup(ctx)
+		if err != nil {
+			return volumetypes.NewTransientOperationFailure(log("mounter.SetUpAt failed to determine if the node service has VOLUME_MOUNT_GROUP capability: %v", err))
+		}
+
+		if driverSupportsCSIVolumeMountGroup {
+			nodePublishFSGroupArg = mounterArgs.FsGroup
+		}
+	}
+
```

Comment:
nit: there is a slight inconsistency in how we handle fsgroup in node-publish and how we handle in node-stage. in node-stage, we check feature-gate inside the CSI client call. Can we keep it uniform? 

Having said that - I know we will have to check for fsgroup support before calling `NodePublish` here, so as fsgroup permissions below can be skipped.

Reply:
Yeah, to make it consistent I was thinking of two options:

1. Add a capabilities check in NodePublishVolume, which then makes a redundant `NodeGetCapabilities` gRPC call
2. Move the capabilities check from NodeStageVolume to MountDevice. I decided against this because relying on the caller (and having to include a comment to describe the behavior) is just slightly more error prone IMO, but I think either option good.

WDYT?

Reply:
I think that - it is caller that should verify feature gate and if plugin supports `VOLUME_MOUNT_GROUP`. We can make a common util function somewhere and call it in both places.

Reply:
Sounds good. I opted not to have a util func because it ended up looking a bit messy: `SetupAt` needs a both a `driverSupportsCSIVolumeMountGroup` bool and an FSGroup arg, whereas `MountDevice` only needs the final arg; I wanted to have a different log message before calls to NodePublish and NodeStage, which would add extra args to the util func.